### PR TITLE
 Restricted possibility to edit profiles via API (Fixed issue #1825) 

### PR DESCRIPTION
**Pull request by petromonka**
dev
## JIRA
 
* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by sharix4 — APPROVED** [review]

File: EPlast/EPlast.Tests/Controllers/UserControllerTests.cs [hunk 1]
```diff
@@ -659,9 +664,61 @@ public async Task EditBase64_ReturnsOkResult()
...
             _mapper.Verify();
             _loggerService.Verify((x) => x.LogInformation(It.IsAny<string>()), Times.Once);
-            Assert.IsInstanceOf<OkResult>(result);
+            Assert.IsInstanceOf<OkObjectResult>(result);
+        }
+
+        [Test]
+        public async Task EditBase64_Returns403Forbidden()
+        {
+            // Assert
```
Comment (christykobyletska) [inline EPlast/EPlast.Tests/Controllers/UserControllerTests.cs:673]: Arrange

File: EPlast/EPlast.Tests/Controllers/UserControllerTests.cs [hunk 2]
```diff
@@ -659,9 +664,61 @@ public async Task EditBase64_ReturnsOkResult()
...
+                .Returns(CreateFakeUser());
+
+            _userManager.Setup(x => x.IsInRoleAsync(It.IsAny<User>(), Roles.Admin)).ReturnsAsync(false);
+            _userService
+                .Setup((x) => x.UpdateAsyncForBase64(It.IsAny<UserDTO>(), It.IsAny<string>(), It.IsAny<int?>(), It.IsAny<int?>(), It.IsAny<int?>(), It.IsAny<int?>()));
+
+            // Act
+            var result = await  _userController.EditBase64(CreateFakeEditUserViewModel());
+            var actual = (result as StatusCodeResult).StatusCode;
+            // Assert
```
Comment (christykobyletska) [inline EPlast/EPlast.Tests/Controllers/UserControllerTests.cs:686]: please add empty line before assert block

File: EPlast/EPlast.Tests/Controllers/UserControllerTests.cs [hunk 3]
```diff
@@ -659,9 +664,61 @@ public async Task EditBase64_ReturnsOkResult()
...
+            _userService.Verify();
+            _mapper.Verify();
+            _loggerService.Verify((x) => x.LogInformation(It.IsAny<string>()), Times.Once);
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        public async Task EditBase64_ReturnsBadRequest()
+        {
+            // Assert
```
Comment (christykobyletska) [inline EPlast/EPlast.Tests/Controllers/UserControllerTests.cs:696]: this block should be named arrange

File: EPlast/EPlast.Tests/Controllers/UserControllerTests.cs [hunk 4]
```diff
@@ -659,9 +664,61 @@ public async Task EditBase64_ReturnsOkResult()
...
+            var result = await _userController.EditBase64(CreateFakeEditUserViewModel());
+
+            // Assert
+            _userService.Verify();
+            _mapper.Verify();
+            _loggerService.Verify((x) => x.LogInformation(It.IsAny<string>()), Times.Once);
+            Assert.IsInstanceOf<BadRequestResult>(result);
         }
 
+
```
Comment (christykobyletska) [inline EPlast/EPlast.Tests/Controllers/UserControllerTests.cs:721]: please delete this unnecessary added empty line

**Review by christykobyletska — CHANGES_REQUESTED** [review]

File: EPlast/EPlast.Tests/Controllers/UserControllerTests.cs [hunk 5]
```diff
@@ -645,10 +646,13 @@ public async Task Edit_Forbidden_Returns403Forbidden()
         public async Task EditBase64_ReturnsOkResult()
         {
             // Assert
```
Comment (christykobyletska) [inline EPlast/EPlast.Tests/Controllers/UserControllerTests.cs:648]: please, change also here to arrange, and look through all test in this file, somewhere could be same situation

**Review by christykobyletska — APPROVED** [review]

**Review by OleksandrBilan — APPROVED** [review]

**Pull request merged**